pkg/controller/plan/adapter/base/network_validation.go

```diff
@@ -0,0 +1,90 @@
+package base
+
+import (
+	api "github.com/kubev2v/forklift/pkg/apis/forklift/v1beta1"
+	"github.com/kubev2v/forklift/pkg/apis/forklift/v1beta1/ref"
+	plancontext "github.com/kubev2v/forklift/pkg/controller/plan/context"
+	liberr "github.com/kubev2v/forklift/pkg/lib/error"
+)
+
+// Network types
+const (
+	Pod     = "pod"
+	Multus  = "multus"
+	Ignored = "ignored"
+)
+
+// NetworkMappingMatcher is a callback function that providers implement
+// to handle their specific network matching logic.
+// It should return true if the mapping applies to the VM's networks.
+// The vm parameter is the VM object retrieved by the provider.
+type NetworkMappingMatcher func(vm interface{}, mapping *api.NetworkPair) (bool, error)
+
+// VMRetriever is a callback function that providers implement
+// to retrieve VM data in their provider-specific way.
+type VMRetriever func(vmRef ref.Ref) (vm interface{}, err error)
```

Comment:
don't like fetching the VM in the validation and then piping it back to the
The validators are meant to do the provider-specific operations